### PR TITLE
Dockerize the scienceapi app

### PR DESCRIPTION
**Pull request by alicoding**

### CODE REVIEW

**Comment by alicoding** [comment]
@gideonthomas

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+app:
+  build: ./scienceapi
```
Comment (gideonthomas) [inline docker-compose.yml]: this is going to be `./app`

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  env_file: .env
+  expose:
+    - "7001"
+  links:
+    - postgres
+  volumes:
+    - ./scienceapi:/app
```
Comment (gideonthomas) [inline docker-compose.yml]: `./app:/app`

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  command: python /app/manage.py runserver 0:7001
+
+nginx:
```
Comment (gideonthomas) [inline docker-compose.yml]: we can remove this whole block

File: nginx/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,7 @@
+FROM nginx:1.9.9
```
Comment (gideonthomas) [inline nginx/Dockerfile]: The whole nginx folder can be removed....we don't need it

File: scienceapi/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FROM python:3.5
```
Comment (gideonthomas) [inline scienceapi/Dockerfile]: Move this file to the newly created `/app` folder

File: scienceapi/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+WORKDIR /app
+EXPOSE 7001
+COPY ./requirements.txt ./
+COPY ./requirements ./requirements
```
Comment (gideonthomas) [inline scienceapi/Dockerfile]: This doesn't need to be here

File: scienceapi/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,7 @@
+FROM python:3.5
+WORKDIR /app
+EXPOSE 7001
+COPY ./requirements.txt ./
```
Comment (gideonthomas) [inline scienceapi/Dockerfile]: This will be `COPY ../requirements.txt ./`

File: scienceapi/manage.py [hunk 8]
```diff
@@ -3,7 +3,7 @@
 import sys
 
 if __name__ == "__main__":
-    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "scienceapi.settings.local")
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "settings.local")
```
Comment (gideonthomas) [inline scienceapi/manage.py]: Undo this change and move the `manage.py` file to the newly created `/app` directory

File: scienceapi/settings/common.py [hunk 9]
```diff
@@ -47,9 +47,9 @@
     'django.contrib.staticfiles',
     'rest_framework',
     'rest_framework_docs',
-    'scienceapi.users',
-    'scienceapi.projects',
-    'scienceapi.events',
+    'users',
```
Comment (gideonthomas) [inline scienceapi/settings/common.py]: Undo these changes

File: scienceapi/settings/common.py [hunk 10]
```diff
@@ -64,7 +64,7 @@
     'whitenoise.middleware.WhiteNoiseMiddleware',
 ]
 
-ROOT_URLCONF = 'scienceapi.urls'
+ROOT_URLCONF = 'urls'
```
Comment (gideonthomas) [inline scienceapi/settings/common.py]: Undo

File: scienceapi/settings/common.py [hunk 11]
```diff
@@ -82,16 +82,23 @@
     },
 ]
 
-WSGI_APPLICATION = 'scienceapi.wsgi.application'
+WSGI_APPLICATION = 'wsgi.application'
```
Comment (gideonthomas) [inline scienceapi/settings/common.py]: Undo

File: scienceapi/settings/common.py [hunk 12]
```diff
@@ -82,16 +82,23 @@
...
 
 
 # Database
 # https://docs.djangoproject.com/en/1.9/ref/settings/#databases
 
 DATABASES = {
-    'default': env.db("DATABASE_URL", default="postgres:///mozillascience"),
+    'default': {
```
Comment (gideonthomas) [inline scienceapi/settings/common.py]: Undo and remove the default value

File: scienceapi/settings/common.py [hunk 13]
```diff
@@ -82,16 +82,23 @@
...
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': os.environ['DB_NAME'],
+        'USER': os.environ['DB_USER'],
+        'PASSWORD': os.environ['DB_PASS'],
+        'HOST': os.environ['DB_HOST'],
+        'PORT': '5432',
+    }
 }
-DATABASES['default']['ATOMIC_REQUESTS'] = True
```
Comment (gideonthomas) [inline scienceapi/settings/common.py]: Undo

File: scienceapi/wsgi.py [hunk 14]
```diff
@@ -11,6 +11,6 @@
 
 from django.core.wsgi import get_wsgi_application
 
-os.environ.setdefault("DJANGO_SETTINGS_MODULE", "scienceapi.settings.production")
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "settings.production")
```
Comment (gideonthomas) [inline scienceapi/wsgi.py]: Undo

File: scienceapi/settings/common.py (hunk 12 above)
Comment (alicoding) [inline scienceapi/settings/common.py]: I think these changes might be okay since we can probably allow this to be configured.

**Comment by alicoding** [comment]
updated.
